File: oc/UIPickerViewDemo/UIPickerViewDemo/FlagView.m
//
//  FlagView.m
//  UIPickerViewDemo
//
//  Created by dfang on 2018-1-9.
//  Copyright © 2018年 east. All rights reserved.
//

#import "FlagView.h"

@implementation FlagView

- (instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        [self addSubview:self.nameLabel];
        [self addSubview:self.iconImageView];
    }
    return self;
}

- (UILabel *)nameLabel {
    if (_nameLabel == nil) {
        _nameLabel = [[UILabel alloc] initWithFrame:CGRectMake(0, 0, 100, self.frame.size.height)];
        _nameLabel.textAlignment = NSTextAlignmentCenter;
    }
    
    return _nameLabel;
}

- (UIImageView *)iconImageView {
    if (_iconImageView == nil) {
        _iconImageView = [[UIImageView alloc] initWithFrame:CGRectMake(150, 0, 100, self.frame.size.height)];
    }
    return _iconImageView;
}

@end

File: swift/UIPickerViewDemo/UIPickerViewDemo/FlagView.swift
//
//  FlagView.swift
//  UIPickerViewDemo
//
//  Created by dfang on 2018-1-5.
//  Copyright © 2018年 east. All rights reserved.
//

import UIKit

class FlagView: UIView {

    let nameLabel = UILabel.init()
    let iconImageView = UIImageView.init()
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        nameLabel.frame = CGRect.init(x: 10, y: 0, width: 80, height: frame.height)
        self.addSubview(nameLabel)
        iconImageView.frame = CGRect.init(x: 90, y: 0, width: 100, height: frame.height)
        self.addSubview(iconImageView)
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}
